src/PublisherServiceProvider.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Plank\Publisher;
+
+use Illuminate\Contracts\Routing\UrlGenerator;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Gate;
+use Plank\Publisher\Commands\PublisherMigrations;
+use Plank\Publisher\Middleware\PublisherMiddleware;
+use Plank\Publisher\Routing\PublisherUrlGenerator;
+use Plank\Publisher\Services\PublisherService;
+use Spatie\LaravelPackageTools\Commands\InstallCommand;
+use Spatie\LaravelPackageTools\Package;
+use Spatie\LaravelPackageTools\PackageServiceProvider;
+
+class PublisherServiceProvider extends PackageServiceProvider
+{
+    public function configurePackage(Package $package): void
+    {
+        $package->name('publisher')
+            ->hasConfigFile()
+            ->hasCommand(PublisherMigrations::class)
+            ->hasInstallCommand(function (InstallCommand $command) {
+                $command->startWith(function (InstallCommand $command) {
+                    $command->line("🖊️  Laravel Publisher Installer... \n");
+
+                    if (! $command->confirm('⚠️ Please ensure you have added the \Plank\Publisher\Contracts\Publishable interface to the models you wish to be publishable. Continue?')) {
+                        throw new \Exception('Installation aborted.');
+                    }
+
+                    if ($command->confirm('Would you like to publish the config file?')) {
+                        $command->publishConfigFile();
+                    }
+
+                    if ($command->confirm('Would you like to add the Publisher migrations?')) {
+                        $command->call('publisher:migrations');
+                        $command->askToRunMigrations();
+                    }
+                });
+
+                $command->endWith(function (InstallCommand $command) {
+                    $command->info('✅ Installation complete.');
+
+                    $command->askToStarRepoOnGitHub('plank/publisher');
```

Comment:
wow thats hilarious, i had no idea this was a function on this class; +10 points for marketing